src/simulation/m_checker.fpp

```diff
@@ -135,6 +135,33 @@ contains
             end if
         end if
 
+        ! phase change checkers. 1st, check if it is activated
+        if ( relax ) then
+            ! 2nd, checking if the correct equation model is chosen
+            if ( model_eqns /= 3 ) then
+                call s_mpi_abort( 'phase change requires model_eqns = 3. ' // &
+                    'Exiting ...')
+            ! if model_eqns == 3, check if the relax models are within the exapected range
+            elseif ( ( relax_model .lt. 0 ) .or. ( relax_model .gt. 6 ) ) then
+                    call s_mpi_abort( 'relax_model should be in between 0 and 6. ' // &
+                    'Exiting ...' )
+            ! if 0 <= relax_model <= 6, check if 0 <= palpha_eps,ptgalpha_eps <= 1d0
+            elseif ( ( palpha_eps .le. 0d0 ) .or. ( palpha_eps .ge. 1d0 ) .or. &
+                   ( ptgalpha_eps .le. 0d0 ) .or. ( ptgalpha_eps .ge. 1d0 ) ) then
+               call s_mpi_abort( 'both palpha_eps and ptgalpha_eps must &
+               be in (0,1). ' // 'Exiting ...')
+           end if
+        elseif ( ( relax_model /= dflt_int ) .or. ( palpha_eps /= 0.0d0 ) &
+            .or. ( ptgalpha_eps /= 0.0d0 ) ) then
+               call s_mpi_abort( 'relax is not set as true, but other phase change parameters have & 
+               been modified. Either activate phase change or set the values to default. ' // 'Exiting ...')
+        end if
+
+        ! note that, for now palpha_eps and ptgalpha_eps have different purposes. In the future,
```

Comment:
these kinds of comments shouldn't be in the code. feel free to keep them in your notes, though.